gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FormConverter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.remote.firestore.schema;
+
+import static com.google.android.gnd.util.ImmutableListCollector.toImmutableList;
+import static java8.util.stream.StreamSupport.stream;
+
+import com.google.android.gnd.model.form.Form;
+
+/** Converts between Firestore nested objects and {@link Form} instances. */
+class FormConverter {
+
+  static Form toForm(String formId, FormNestedObject obj) {
+    return Form.newBuilder()
+        .setId(formId)
```

Comment:
Is there a reason we don't store formIds on forms themselves?

Reply:
Because they come from the enclosing map, e.g.
```
{
   'id1234': {
     // nested object
   }
}
```

Reply:
I see, thanks!